### PR TITLE
refactor: use `google-cloud-cpp-core` feedstock

### PR DESCRIPTION
**Pull request by coryan**
Start using the packages created by `google-cloud-cpp-core-feedstock`. That requires an upgrade to v2.17.0.

* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* N/A: Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Part of the work for #141

### CODE REVIEW

**Comment by coryan** [comment]
FWIW, I tested locally with:

```shell
~/miniconda3/bin/conda smithy rerender -c auto
```

And it completed successfully:

```log
INFO:conda_smithy.configure_feedstock:Downloading conda-forge-pinning-2023.12.19.13.04.28
INFO:conda_smithy.configure_feedstock:Extracting conda-forge-pinning to /usr/local/google/home/coryan/.cache/conda-smithy
INFO:conda_smithy.configure_feedstock:__pycache__ rendering is skipped
INFO:conda_smithy.configure_feedstock:README rendering is skipped
WARNING: Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
WARNING: Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
WARNING: No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.22
Adding in variants from internal_defaults
Adding in variants from /usr/local/google/home/coryan/.cache/conda-smithy/conda_build_config.yaml
Adding in variants from argument_variants
WARNING:conda_smithy.configure_feedstock:Azure build_id can't be retrieved using the Azure token. Exception: list index out of range
INFO:conda_smithy.configure_feedstock:Re-rendered with conda-build 3.27.0, conda-smithy 3.30.1, and conda-forge-pinning 2023.12.19.13.04.28
[main 4fb81fa] MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.30.1, and conda-forge-pinning 2023.12.19.13.04.28
 22 files changed, 1075 insertions(+)
 create mode 100755 .azure-pipelines/azure-pipelines-linux.yml
 create mode 100755 .azure-pipelines/azure-pipelines-osx.yml
 create mode 100755 .azure-pipelines/azure-pipelines-win.yml
 create mode 100644 .ci_support/README
 create mode 100644 .ci_support/linux_64_.yaml
 create mode 100644 .ci_support/osx_64_.yaml
 create mode 100644 .ci_support/win_64_.yaml
 create mode 100644 .circleci/config.yml
 create mode 100644 .gitattributes
 create mode 100644 .github/CODEOWNERS
 create mode 100644 .github/workflows/automerge.yml
 create mode 100644 .github/workflows/webservices.yml
 create mode 100644 .gitignore
 create mode 100755 .scripts/build_steps.sh
 create mode 100644 .scripts/logging_utils.sh
 create mode 100755 .scripts/run_docker_build.sh
 create mode 100755 .scripts/run_osx_build.sh
 create mode 100755 .scripts/run_win_build.bat
 create mode 100644 LICENSE.txt
 create mode 100644 README.md
 create mode 100644 azure-pipelines.yml
 create mode 100755 build-locally.py
INFO:conda_smithy.configure_feedstock:
INFO:conda_smithy.configure_feedstock:These changes need to be pushed to github!
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -56,87 +51,72 @@ outputs:
         - ninja
         - libgrpc
         - libprotobuf
+        - libgoogle-cloud-devel =={{ version }}
       host:
         - libabseil
-        - libcrc32c
-        - libcurl
         - libgrpc
         - libprotobuf
```
Comment (h-vetinari) [inline recipe/meta.yaml:58]: `libgoogle-cloud-all` will IMO need to pull in all the outputs from google-cloud-cpp-core:
```suggestion
        - libprotobuf
      run:
        - libgoogle-cloud-bigtable
        # not yet available
        # - libgoogle-cloud-compute
        # - libgoogle-cloud-compute-ai
        - libgoogle-cloud-iam
        - libgoogle-cloud-oauth2
        - libgoogle-cloud-policytroubleshooter
        - libgoogle-cloud-pubsub
        - libgoogle-cloud-spanner
        - libgoogle-cloud-storage
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -56,87 +51,72 @@ outputs:
...
-        - openssl
-        - nlohmann_json
+        - libgoogle-cloud-devel =={{ version }}
       run:
-        - {{ pin_subpackage("libgoogle-cloud", exact=True) }}
-        - nlohmann_json
-        # cannot load package without zlib because it's specified
-        # as required; if possible, this should be removed upstream
-        - zlib
+        - {{ pin_subpackage("libgoogle-cloud-all", exact=True) }}
```
Comment (h-vetinari) [inline recipe/meta.yaml:95]: Same for all the devel packages here

File: recipe/meta.yaml (hunk 1 above)
Comment (coryan) [reply recipe/meta.yaml:58]: Fixed.

File: recipe/meta.yaml (hunk 2 above)
Comment (coryan) [reply recipe/meta.yaml:117]: Fixed.

**Comment by coryan** [comment]
Back to draft or I will accidentally push the merge button.

**Comment by coryan** [comment]
@h-vetinari this is looking good now. Do you mind going over it once more to check my work?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -29,114 +21,133 @@ requirements:
...
```
Comment (h-vetinari) [inline recipe/meta.yaml:106]: I think we need to specify the `*-devel` packages here for all the other packages being built in other feedstocks, no?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -29,114 +21,133 @@ requirements:
...
+        # From the google-cloud-cpp-compute feedstock
+        - libgoogle-cloud-compute-devel
+        # From the google-cloud-cpp-ai feedstock
+        - libgoogle-cloud-aiplatform-devel
+        - libgoogle-cloud-automl-devel
+        - libgoogle-cloud-discoveryengine-devel
+        - libgoogle-cloud-dialogflow_es-devel
+        - libgoogle-cloud-dialogflow_cx-devel
+        - libgoogle-cloud-dlp-devel
+        - libgoogle-cloud-speech-devel
```
Comment (h-vetinari) [inline recipe/meta.yaml:143]: I.e. this block should be a run-dep of `libgoogle-cloud-all-devel`, and then we don't need it as a test dep.

**Review by h-vetinari — COMMENTED** [review]
Basically LGTM, glad we're finally in the home stretch here! 🥳 

Just some last remaining dependency specification issues.

File: recipe/meta.yaml (hunk 4 above)
Comment (coryan) [reply recipe/meta.yaml:127]: Doh! Fixed.

File: recipe/meta.yaml (hunk 3 above)
Comment (coryan) [reply recipe/meta.yaml:106]: Done

File: recipe/build.sh [hunk 5]
```diff
@@ -9,18 +9,49 @@ if [[ "${target_platform}" == osx-* ]]; then
...
-    -DCMAKE_BUILD_TYPE=release \
-    -DCMAKE_CXX_STANDARD=17 \
-    -DCMAKE_INSTALL_PREFIX=$PREFIX \
-    -DCMAKE_INSTALL_LIBDIR=lib \
-    -DProtobuf_PROTOC_EXECUTABLE=$BUILD_PREFIX/bin/protoc \
-    -DGOOGLE_CLOUD_CPP_GRPC_PLUGIN_EXECUTABLE=$BUILD_PREFIX/bin/grpc_cpp_plugin \
-    -DGOOGLE_CLOUD_CPP_ENABLE_WERROR=OFF
+# This list represents some of the larger features in `google-cloud-cpp`.
+# Moving them to their own feedstock splits the build time across feedstocks.
+# The features happen to be AI and ML related, which provides a good nemonic
```
Comment (h-vetinari) [inline recipe/build.sh:14]:
```suggestion
# The features happen to be AI and ML related, which provides a good mnemonic
```

**Review by h-vetinari — APPROVED** [review]
🥳

File: recipe/build.sh (hunk 5 above)
Comment (coryan) [reply recipe/build.sh]: Done.

**Comment by coryan** [comment]
> Thank you for your patience and persistence on this one

I feel I should be thanking you for staying with me through the project.  I look forward to upgrading to v2.21 once this PR goes through.

**Comment by coryan** [comment]
The failure is very odd.  Seems like a flake:

- The local build on macOS + arm64 succeeded.
- The error messages make no sense for a C++ build

I will try closing and reopening this PR.

**Comment by h-vetinari** [comment]
> I will try closing and reopening this PR.

It's better to just do a targeted restart of the failed builds in the GH UI.

**Comment by h-vetinari** [comment]
> The error messages make no sense for a C++ build

Indeed, it kinda looks like it tried to read a template file (or comment?) as a C++ file. Perhaps we managed to hit some weird filesystem corruption or so... Let's see what happens with the rerun.

**Comment by h-vetinari** [comment]
It happened again after merging, this time with an even weirder error (that makes it seem like some sort of corruption happening even more):
```
FAILED: google/cloud/beyondcorp/CMakeFiles/google_cloud_cpp_beyondcorp_protos.dir/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc.o 
$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-clang++ -DPROTOBUF_USE_DLLS -Dgoogle_cloud_cpp_beyondcorp_protos_EXPORTS -isystem $SRC_DIR/build/google/cloud/beyondcorp -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/google-cloud-cpp-split-2.17.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D_LIBCPP_DISABLE_AVAILABILITY -O3 -DNDEBUG -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -mmacosx-version-min=11.0 -fPIC -MD -MT google/cloud/beyondcorp/CMakeFiles/google_cloud_cpp_beyondcorp_protos.dir/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc.o -MF google/cloud/beyondcorp/CMakeFiles/google_cloud_cpp_beyondcorp_protos.dir/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc.o.d -o google/cloud/beyondcorp/CMakeFiles/google_cloud_cpp_beyondcorp_protos.dir/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc.o -c $SRC_DIR/build/google/cloud/beyondcorp/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc
In file included from $SRC_DIR/build/google/cloud/beyondcorp/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.cc:4:
$SRC_DIR/build/google/cloud/beyondcorp/google/cloud/beyondcorp/appconnectors/v1/app_connectors_service.pb.h:1:1: error: unknown type name 'iYbWe6HlW1wDDfP7aKYT1XHcqQIE'
```

In particular: `unknown type name 'iYbWe6HlW1wDDfP7aKYT1XHcqQIE'` 😂

I'll restart the build as soon as the other jobs have finished

**Pull request merged**